extensions/cassandra-storage/src/main/java/io/druid/storage/cassandra/CassandraDataSegmentPuller.java

```diff
@@ -26,70 +26,76 @@
 import io.druid.timeline.DataSegment;
-import io.druid.utils.CompressionUtils;
 import org.apache.commons.io.FileUtils;
 
 import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
 import java.io.OutputStream;
 
 /**
  * Cassandra Segment Puller
- *
- * @author boneill42
  */
 public class CassandraDataSegmentPuller extends CassandraStorage implements DataSegmentPuller
 {
-	private static final Logger log = new Logger(CassandraDataSegmentPuller.class);
-	private static final int CONCURRENCY = 10;
-	private static final int BATCH_SIZE = 10;
+  public static final String SCHEME = "cassandra";
+  private static final Logger log = new Logger(CassandraDataSegmentPuller.class);
+  private static final int CONCURRENCY = 10;
+  private static final int BATCH_SIZE = 10;
 
   @Inject
-	public CassandraDataSegmentPuller(CassandraDataSegmentConfig config)
-	{
-		super(config);
-	}
-
-	@Override
-	public void getSegmentFiles(DataSegment segment, File outDir) throws SegmentLoadingException
-	{
-		String key = (String) segment.getLoadSpec().get("key");
-		log.info("Pulling index from C* at path[%s] to outDir[%s]", key, outDir);
+  public CassandraDataSegmentPuller(CassandraDataSegmentConfig config)
+  {
+    super(config);
+  }
 
-		if (!outDir.exists())
-		{
-			outDir.mkdirs();
-		}
+  @Override
+  public void getSegmentFiles(DataSegment segment, File outDir) throws SegmentLoadingException
+  {
+    String key = (String) segment.getLoadSpec().get("key");
+    getSegmentFiles(key, outDir);
+  }
+  public Long getSegmentFiles(String key, File outDir) throws SegmentLoadingException{
+    log.info("Pulling index from C* at path[%s] to outDir[%s]", key, outDir);
+    if (!outDir.exists()) {
+      outDir.mkdirs();
+    }
 
-		if (!outDir.isDirectory())
-		{
-			throw new ISE("outDir[%s] must be a directory.", outDir);
-		}
+    if (!outDir.isDirectory()) {
+      throw new ISE("outDir[%s] must be a directory.", outDir);
+    }
 
-		long startTime = System.currentTimeMillis();
-		ObjectMetadata meta = null;
-		final File outFile = new File(outDir, "index.zip");
-		try
-		{
-			try
-			{
-				log.info("Writing to [%s]", outFile.getAbsolutePath());
-				OutputStream os = Files.newOutputStreamSupplier(outFile).getOutput();
-				meta = ChunkedStorage
-				    .newReader(indexStorage, key, os)
-				    .withBatchSize(BATCH_SIZE)
-				    .withConcurrencyLevel(CONCURRENCY)
-				    .call();
-				os.close();
-				CompressionUtils.unzip(outFile, outDir);
-			} catch (Exception e)
-			{
-				FileUtils.deleteDirectory(outDir);
-			}
-		} catch (Exception e)
-		{
-			throw new SegmentLoadingException(e, e.getMessage());
-		}
-		log.info("Pull of file[%s] completed in %,d millis (%s bytes)", key, System.currentTimeMillis() - startTime,
-		    meta.getObjectSize());
-	}
+    long startTime = System.currentTimeMillis();
+    ObjectMetadata meta = null;
+    final File tmpFile = new File(outDir, "index.zip");
+    tmpFile.deleteOnExit();
```

Comment:
`tmpFile.deleteOnExit()` registers the file with a shutdown listener that I'm pretty sure will exist until the process dies.  I.e. for a really long-running process, this is a memory leak.  It might be a slow trickle, but still likely a leak.  Can we make sure we delete the thing using a finally block or something instead?